### DISCUSSION TITLE
EPERM: operation not permitted, copyfile - when working on files that reside on google drive map

### DISCUSSION THREAD

**Discussion by dss010101** [devcontainers/community#139] [General]
Im working in a devcontainer container (debian 12 bookworm).  
The container builds fine and i can run my code.
However, if try to copy a file in the vscode explorer and paste to another folder i end up with errors like this
My files are on google drive. I have the google drive desktop installed which is the G drive on my local.
Furthermore i have the folders marked as "available offline"

```
The file(s) to paste have been deleted or moved since you copied them. 
Error: EPERM: operation not permitted, copyfile '/workspaces/my_app/src/my_app/__init__.py' -> '/workspaces/my_app/src/my_app/web/__init__.py'
```

I have googled and found recommendation about running vscode as administrator, but this does not help.
As mentioned, the folders and file are local (available locally).
I have also made sure all permissions are read/write.

Any other suggestions to solve this issue?

**Comment by bamurtaugh** [comment]
Thanks for opening. @chrmarti this sounds like it'd be along the lines of the request in https://github.com/microsoft/vscode-remote-release/issues/1411 - what do you think?

**Reply by chrmarti** [reply]
This seems to be about copying a file in the VS Code Explorer and pasting it in another folder in the VS Code Explorer.

@bpasero If the Explorer can access the file, what might prevent it from copying it to another folder? The error sounds like the copy operation is using another way of accessing it than the Explorer when just listing the folder content.

**Reply by bpasero** [reply]
We are not doing anything special other than `fs.copyFile` recursively:

https://github.com/microsoft/vscode/blob/403294d92b4fbcdad61bb74635a8e5e145151aaa/src/vs/base/node/pfs.ts#L630-L649
